src/chat/ChatScreen.js

```diff
@@ -32,16 +34,35 @@ type Props = $ReadOnly<{|
   ...SelectorProps,
 |}>;
 
-class ChatScreen extends PureComponent<Props> {
+type State = {|
+  editMessage: EditMessage | null,
+|};
+
+class ChatScreen extends PureComponent<Props, State> {
   context: Context;
 
+  state = {
+    editMessage: null,
+  };
+
   static contextTypes = {
     styles: () => null,
   };
 
+  onEditStart = async (editMessage: EditMessage) => {
+    this.setState({
+      editMessage,
+    });
+  };
+
+  onEditComplete = () => {
+    this.setState({ editMessage: null });
+  };
+
   render() {
     const { styles: contextStyles } = this.context;
     const { fetching, haveNoMessages, loading, navigation } = this.props;
+    const { editMessage } = this.state;
     const { narrow } = navigation.state.params;
```

Comment:
nit: the line getting stuff from out of `navigation` is directly connected to the line that got `navigation` from out of `this.props`, so natural to keep them right next to each other